scripts/commands/checkOrphanPages.ts

```diff
@@ -0,0 +1,147 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import fs from "fs/promises";
+
+import { globby } from "globby";
+import yargs from "yargs/yargs";
+import { hideBin } from "yargs/helpers";
+import { flattenDeep } from "lodash";
+
+interface Arguments {
+  [x: string]: unknown;
+  currentApis: boolean;
+  devApis: boolean;
+  historicalApis: boolean;
+}
+
+const ALLOWED_ORPHANS = new Set([
+  "/api/qiskit/qiskit.primitives.BaseEstimator",
+  "/api/qiskit/qiskit.primitives.BaseSampler",
+]);
+
+const readArgs = (): Arguments => {
+  return yargs(hideBin(process.argv))
+    .version(false)
+    .option("current-apis", {
+      type: "boolean",
+      default: false,
+      description: "Check the links in the current API docs.",
+    })
+    .option("dev-apis", {
+      type: "boolean",
+      default: false,
+      description: "Check the links in the /dev API docs.",
+    })
+    .option("historical-apis", {
+      type: "boolean",
+      default: false,
+      description:
+        "Check the links in the historical API docs, e.g. `api/qiskit/0.44`. ",
+    })
+    .parseSync();
+};
+
+async function main() {
+  const args = readArgs();
+
+  const tocFiles = await determineTocFiles(args);
+
+  let allGood = true;
+  const orphanPages = [];
+
+  for (tocFile of tocFiles) {
+    tocUrls = await getTocUrls(tocFile);
+    const dir = tocFile.replace("_toc.json", "");
+    const dirFiles = await collectDirFiles(dir);
+    for (file of dirFiles) {
+      if (!tocUrls.has(file) && !ALLOWED_ORPHANS.has(file)) {
+        allGood = false;
+        orphanPages.push(file);
+      }
+    }
+  }
+  if (!allGood) {
+    console.error(
+      "\n There's some orphaned pages!  These files need a home: \n",
+      orphanPages,
+    );
+    process.exit(1);
+  }
+  console.log("\nNo orphan pages found ✅\n");
+}
+
+async function getTocUrls(filePath: string): Set<string> {
+  console.log("Checking toc in:", tocFile);
```

Comment:
Nit: I recommend moving this log back to the `main` function so that the control flow is more clear. This log is referring to the entire process of checking a folder, whereas this function is solely about reading the single toc.json as step 1. Since the log refers to the entire process, it's better to live at that abstraction level rather than hidden inside a single step.